source/style-guide/style/placeholder-variable-text.txt

```diff
@@ -26,13 +28,79 @@ name of a directory.
    the path. For information about formatting environment variables,
    see :ref:`text-formatting`.
 
+Placeholder Text Guidelines
+---------------------------
+
 When creating placeholder text, use the following guidelines.
 
+Apply Language-Specific Formatting
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :widths: 20 20 60
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Language
+     - Format
+     - Additional Guidelines
+
+   * - Shell, PowerShell, Command Prompt
+     - ``<variable-name>``
+     - Follow `docopt <http://docopt.org/>`__
+       guidelines
+
+   * - API
+     - ``{variableName}``
+     - Follow the
+       `OpenAPI spec <https://swagger.io/docs/specification/describing-parameters/>`__
+
+   * - Java
+     - ``<variable_name>``, ``<class_data_member_>``
```

Comment:
The second item is a little confusing to me. Can you give me an example of what a class data member is? And is this suggesting it should contain a trailing underscore?

Reply:
OK, this is a mistake. I'm trying to base this on [Google's style guide](https://google.github.io/styleguide/javaguide.html#s5-naming).

Reply:
That would be, in detail:

| Variable Type | Example |
| --- | --- |
| package names | com.example.packagename |
| class names | ClassName |
| method names | methodName |
| constant names | CONSTANT_NAME |
| non-constant field names | nonConstantFieldName |
| parameter names | parameterName |
| type variable names | T or TypeT |